Signal/src/call/SignalCall.swift

```diff
@@ -39,51 +42,97 @@ protocol CallObserver: class {
 
     // Distinguishes between calls locally, e.g. in CallKit
     let localId: UUID
-    
+
     var hasVideo = false {
         didSet {
-            Logger.debug("\(TAG) hasVideo changed: \(oldValue) -> \(hasVideo)")
-            for observer in observers {
-                observer.value?.hasVideoDidChange(call: self, hasVideo: hasVideo)
+            // This should only occur on the signaling queue.
+            objc_sync_enter(self)
```

Comment:
SignalCall state is accessed and its methods called from more than one thread, not just the signaling queue. I'd like to propose that we try to move SignalCall to being accessed only on the main thread. Until that time, I've protected all of its state through synchronization.

More generally, I'd like to propose that we have an explicit understanding that all state should only be accessed from one thread. 

I'd also like to propose that signaling queue become a private implementation detail of ServiceCall.  

Looking forward to hearing your thoughts about this.